### DISCUSSION TITLE
Works with macOS?

### DISCUSSION THREAD

**Discussion by dwormuth** [rustprooflabs/pgdd#51] [Q&A]
Specifically Sonoma?

**Comment by rustprooflabs** [comment]
Good question, I don't have a way to test that myself. My next computer might be a Mac, but don't have one at this time to test on.

Per the [pgrx project](https://github.com/pgcentralfoundation/pgrx/tree/17506399ce482c1ab65b087fca7dd57d81cf5b37?tab=readme-ov-file#system-requirements) it does say MacOS is supported.

> "PGRX has been tested to work on x86_64⹋ and aarch64⹋ Linux and aarch64 **macOS** targets. "

It should be possible to use `pgrx` to build on MacOS, the instructions in the [PgDD Docs](https://rustprooflabs.github.io/pgdd/create-installer.html#create-binary-installer-wout-docker) explain the steps for Ubuntu, not sure if that helps at all. The [pgrx docs](https://github.com/pgcentralfoundation/pgrx/tree/17506399ce482c1ab65b087fca7dd57d81cf5b37?tab=readme-ov-file#system-requirements) have more details about MacOS.

**Comment by dwormuth** [comment]
I see binaries to download, but no "rust" build instructions. Cargo pgrx
run seems pretty generic to build pgdd. Perhaps I'm missing something.

Dave

**Comment by dwormuth** [comment]
Any thoughts on creating a section like "Create a Package" but instead
"Create a Binary -- Or -- Compile the Extension?"

I'm not experienced enough to read your hints and see what is actually
being done with the packaging scripts. I could then move the extension into
the PostgeSQL tree to enjoy your extension.

Might help the windows folks too.

Dave

**Reply by rustprooflabs** [reply]
The section [Create Binary Installer w/out Docker](https://rustprooflabs.github.io/pgdd/create-installer.html#create-binary-installer-wout-docker) linked in my prior comment has the instructions you're asking for.  `cargo pgrx package` is the primary command to create the binaries. The `run` subcommand is mostly helpful for development/testing. The steps installing repreqs for `pgrx` are the most likely places to deviate, but there are likely other subtle differences too. I have no way to test them on MacOS to make them more accurate. If you can document specific differences for MacOS that would be a PR I'd be happy to review.

Windows isn't supported by pgrx per their [Caveats and Known issues](https://github.com/pgcentralfoundation/pgrx/tree/17506399ce482c1ab65b087fca7dd57d81cf5b37?tab=readme-ov-file#caveats--known-issues): "Windows is not supported."
